CMakeLists.txt

```diff
@@ -262,6 +262,14 @@ set_target_properties(datachannel PROPERTIES
 	SOVERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}
 	CXX_STANDARD 17
 	CXX_VISIBILITY_PRESET default)
+
+if(APPLE)
+	set_target_properties(datachannel PROPERTIES
+		SOVERSION ${PROJECT_VERSION}
```

Comment:
Shouldn't you override VERSION to `0` here?

Reply:
👍 

So this PR doesn't follow the suggestions from [here](https://github.com/obsproject/obs-deps/pull/204#discussion_r1449211126). When I apply those changes it breaks the tests. I worry it will break others that depend on libdatachannel also.

The PR in its current from is just reverting 7591b9695c partially.